### PR TITLE
chore: Added the new Parameter of the generator to the CLI

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Proposed Changes

I added the new parameters to the CLI. This will become all redundant once we merged the two repos. Then just one CLI interface is needed.

## Checklist

- [x] I have added or adjusted tests that prove my fix is effective or that my feature works
- [x] I have added or adjusted documentation

### CODE REVIEW

File: src/utils/generate-odata-client-util.ts [hunk 1]
```diff
@@ -33,25 +34,47 @@ type AllOptions = GeneratorOptionsSDK & GeneratorOptionCli;
...
+  [optionName in NumberArgKeys]: IOptionFlag<number | undefined>;
+};
+
 export type FlagsParsed = {
   [Key in keyof AllOptions]: AllOptions[Key] extends boolean
     ? boolean
     : string | undefined;
 };
 
+export function toIntegerFlag(yargsNumber: Options): IOptionFlag<number | undefined> {
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply src/utils/generate-odata-client-util.ts:69]: We found that in the renaming from master -> main a few commits were lost. In the process of regaining them and merging main I meesed up a bit. I will write once all changes are back in.

**Comment by FrankEssenberger** [comment]
@jjtang1985 @marikaner now you can review after the main/master issue is resolved.

File: src/commands/generate-odata-client.ts [hunk 2]
```diff
@@ -43,6 +43,12 @@ export default class GenerateODataClient extends Command {
     sdkAfterVersionScript: toBooleanFlag(
       generatorOptionsSDK.sdkAfterVersionScript
     ),
+    versionInPackageJson: toStringFlag(
+      generatorOptionsSDK.versionInPackageJson!
+    ),
+    processesJsGeneration: toIntegerFlag(
+      generatorOptionsSDK.processesJsGeneration!
```
Comment (jjtang1985) [inline src/commands/generate-odata-client.ts:50]: This is optional, but with the default value, it's fine to use `!`.

File: src/commands/generate-odata-client.ts [hunk 3]
```diff
@@ -43,6 +43,12 @@ export default class GenerateODataClient extends Command {
     sdkAfterVersionScript: toBooleanFlag(
       generatorOptionsSDK.sdkAfterVersionScript
     ),
+    versionInPackageJson: toStringFlag(
+      generatorOptionsSDK.versionInPackageJson!
```
Comment (jjtang1985) [inline src/commands/generate-odata-client.ts:47]: This is optional parameter, without default value, we should consider null check.

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
Look good, only one comment about `versionInPackageJson!`.

File: src/commands/generate-odata-client.ts (hunk 3 above)
Comment (FrankEssenberger) [reply src/commands/generate-odata-client.ts:47]: the `generatorOptionsSDK` is an object instance just mapping the name of the variable `versionInPackageJson` to the options in `yargs` like short name, type etc. So this is always there. All options are described there and not null. The interface `GeneratorOptionsSDK` is then used at runtime and here some keys could be non optional.  

However, I found that with `-?` you can remove the modifier from the `keyof` so we can get rid of all the `!`.

**Comment by FrankEssenberger** [comment]
@jjtang1985 can you have a look if it is ok now?

File: src/commands/generate-odata-client.ts (hunk 3 above)
Comment (jjtang1985) [reply src/commands/generate-odata-client.ts]: Maybe you can share the doc regarding `-?`

**Review by jjtang1985 — APPROVED** [review]

**Review by marikaner — APPROVED** [review]
LGTM

**Pull request merged**